config/main.py

```diff
@@ -673,8 +685,12 @@ def _stop_services():
         pass
 
     click.echo("Stopping SONiC target ...")
-    clicommon.run_command("sudo systemctl stop sonic.target")
-
+    (out, err, rv) = clicommon.run_command("sudo systemctl stop sonic.target", return_err=True)
+    if rv != 0:
+        click.echo("Failed to stop sonic.target rv={}.".format(rv))
+        log.log_notice("Failed to stop sonic.target rv={} stdout='{}' stderr='{}'.".format(rv, out.rstrip('\n'), err.rstrip('\n')))
+        click.echo("Stopping all services individually.")
+        _stop_all_services()
 
 def _get_sonic_services():
     out = clicommon.run_command("systemctl list-dependencies --plain sonic.target | sed '1d'", return_cmd=True)
```

Comment:
>systemctl list-dependencies --plain [](http://example.com/codeflow?start=33&length=35)

Is it a general solution? If in future there is multiple level of dependencies as a tree, will this command still work?

Reply:
I think that's why sonic.target was introduced, if some docker-startup service was to be added which, it should according to the current design have a WantedBy Dependency on the sonic.target and it'll then be listed by this line

@stepanblyschak, can you provide more info on this?

Reply:
Is it same as `systemctl stop sonic.target` twice? If yes, this may be more general solution.

Reply:
Not Exactly and as i've said earlier, Stopping services individually will reduce the likelihood of race condition around conflicting job installations. "sonic.target" is is very sensitive to the conflicting jobs for any of the dependent services. And a stop job for sonic.target takes around 1-2 sec. This entire period of time is prone to race conditions. 

On the other hand stopping individual services would mean that the likelihood of race condition significantly reduces. So, i believe, this is more robust than just stopping sonic.target twice

Reply:
Sorry for late comment. 

> Stopping services individually will reduce the likelihood of race condition around conflicting job installations. "sonic.target" is is very sensitive to the conflicting jobs for any of the dependent services

If `sonic.target` is not robust solution to stop all the services, what is the point to introduce it at first? Should we stop using it in `config` implementation completely, and use only the method of "Stopping services individually"? and since this method is robust, we only need it once.

Reply:
Hmm makes sense, we can infact remove the sonic.target stop job